airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/streams/common.py

```diff
@@ -0,0 +1,255 @@
+accomlished by single http call (any but report streams). Also in contains
+transform method to modify output data (see transform method comment).
+
+SubProfilesStream is subclass for http streams to perform read_records from
+basic class for EACH profile from self._profiles list.
+
+PaginationStream class implement Amazon Ads API pagintaion for all requests.
+This is base class for all the sync http streams that used by source.
+
+ReportStream (It implemented on report_stream.py file) is subclass for async
+report streams. This is not standard http stream and used for generating
+reports for profiles from BasicAmazonAdsStream _profiles list.
+
+"""
+
+
+class ErrorResponse(BaseModel):
+    code: str
+    details: str
+    requestId: str
+
+
+class BasicAmazonAdsStream(Stream, ABC):
+    """
+    Base class for all Amazon Ads streams.
+    """
+
+    def __init__(self, config, profiles: List[Profile] = None):
+        self._profiles = profiles or []
+        self._client_id = config.client_id
+        self._url = config.host or URL_BASE
+
+    @property
+    @abstractmethod
+    def model(self) -> CatalogModel:
+        """
+        Pydantic model to represent json schema
+        """
+
+    def get_json_schema(self):
+        return self.model.schema()
+
+
+# Basic full refresh stream
+class AmazonAdsStream(HttpStream, BasicAmazonAdsStream):
+    """
+    Class for getting data from streams that based on single http request.
+    """
+
+    # List of properties that used by transform method to convert "object"
+    # subtype with complex dict to plan string.
+    flatten_properties = []
+
+    def __init__(self, config, *args, profiles: List[Profile] = None, **kwargs):
+        BasicAmazonAdsStream.__init__(self, config, profiles=profiles)
+        HttpStream.__init__(self, *args, **kwargs)
+
+    @property
+    def url_base(self):
+        return self._url
+
+    def transform(self, record: dict):
+        """
+        Convert properties that listed on flatten_properties variable from dict
+        to plain string. We need it to not overcomplicate catalog model with
+        objects containing internal information and unlikely to be used for analytics.
+        """
+        for prop in self.flatten_properties:
+            if prop in record:
+                record[prop] = json.dumps(record[prop])
+        return record
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def request_headers(self, *args, **kvargs) -> MutableMapping[str, Any]:
+        return {"Amazon-Advertising-API-ClientId": self._client_id}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        """
+        :return an object representing single record in the response
+        """
+        yield from response.json() if response else []
+
+    def _send_request(self, request: requests.PreparedRequest, request_kwargs: Mapping[str, Any]) -> requests.Response:
```

Comment:
we should really not override `_` methods in the CDK because they can change without any notice and break consumers like this one. I _think_ you should be able to use [this PR](https://github.com/airbytehq/airbyte/pull/5363) once it's released. If you're blocked on that PR, just assume it has been merged and build off of that (it should be merged very very soon)

Reply:
Done